### PR TITLE
Skeleton for view

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 case class NewPageContext(
-  keeps: Seq[NewKeepInfo],
+  keeps: Seq[RestrictedKeepInfo], // TODO(ryan): should this switch to NewKeepInfo?
   numVisibleKeeps: Int,
-  numTotalKeeps: Int,
-  sources: Seq[SourceAttribution],
-  numVisibleSources: Int,
-  numTotalSources: Int,
+  // numTotalKeeps: Int, // TODO(ryan): is it even possible to retrive this information? LimitedAugmentationInfo doesn't have it?
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala]: Yes

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+  // sources: Seq[SourceAttribution], // TODO(ryan): figure out how to retrieve
+  // numVisibleSources: Int,
+  // numTotalSources: Int,
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala:12]: `relevant` is probably more accurate than `visible`. If something's not visible you won't even know about it. This is more about what we deem worth returning.

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala [hunk 3]
```diff
@@ -1,15 +1,17 @@
 package com.keepit.shoebox.data.keep
 
 import com.keepit.model.{ Hashtag, BasicLibrary, SourceAttribution }
+import com.keepit.search.augmentation.{ RestrictedKeepInfo, LimitedAugmentationInfo }
 import com.keepit.social.BasicUser
+import play.api.libs.json.{ Writes, Json }
 
 case class NewPageContext(
-  keeps: Seq[NewKeepInfo],
+  keeps: Seq[RestrictedKeepInfo], // TODO(ryan): should this switch to NewKeepInfo?
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala]: Probably, but that requires some work and opening up some of our data in search (i.e. redefining what `discoverable` means)

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala]: I think we can construct the keep infos given their ids, but it's more expensive.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContext.scala]: To expand on this (I thought there was a task for it): currently we have keeps that are "partially" viewable, i.e. keeps in `Main` let your network know they exist, but people can't actually get to them. This is why `RestrictedKeepInfo` exists.

**Pull request merged**